src/main/java/io/scim2/swagger/client/ScimApiClient.java

```diff
@@ -105,17 +137,22 @@ public class ScimApiClient {
     private boolean verifyingSsl;
     private KeyManager[] keyManagers;
 
-    private OkHttpClient httpClient;
+    private HttpClient httpClient;
+    private final HttpClientBuilder httpClientBuilder;
+    private RequestConfig requestConfig;
```

Comment:
Is there any special reason to keep `httpClientBuilder` and `requestConfig` as class level attributes? Feels like, we should define these inside the method level when initiating the `httpClient`

Reply:
These were introduced as class level attributes to support setting and getting some configurations which cannot be set or retrieved directly from the `httpClient` object in the Apache library.
`requestConfig` - used in `getConnectTimeout` method to get the timeout value being set in `setConnectTimeout` method.
`httpClientBuilder` - used for setting configs like setting timeout, setting SSLHostnameVerifier. HttpClient is built after setting these. 

Class level variables are required to support these settings being set in different methods.